guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryViewImpl.java

```diff
@@ -0,0 +1,322 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.structure.client.editors.repository.clone;
+
+import javax.enterprise.context.Dependent;
+
+import com.github.gwtbootstrap.client.ui.Button;
+import com.github.gwtbootstrap.client.ui.ControlGroup;
+import com.github.gwtbootstrap.client.ui.HelpInline;
+import com.github.gwtbootstrap.client.ui.ListBox;
+import com.github.gwtbootstrap.client.ui.PasswordTextBox;
+import com.github.gwtbootstrap.client.ui.TextBox;
+import com.github.gwtbootstrap.client.ui.constants.ControlGroupType;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.KeyPressEvent;
+import com.google.gwt.event.dom.client.KeyPressHandler;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.InlineHTML;
+import com.google.gwt.user.client.ui.PopupPanel;
+import com.google.gwt.user.client.ui.Widget;
+import org.guvnor.structure.client.resources.i18n.CommonConstants;
+import org.uberfire.ext.widgets.common.client.common.BusyPopup;
+import org.uberfire.ext.widgets.common.client.common.popups.BaseModal;
+import org.uberfire.ext.widgets.common.client.common.popups.errors.ErrorPopup;
+import org.uberfire.ext.widgets.core.client.resources.i18n.CoreConstants;
+
+@Dependent
+public class CloneRepositoryViewImpl
+        extends PopupPanel implements CloneRepositoryView {
+
+    interface CloneRepositoryFormBinder
+            extends
+            UiBinder<Widget, CloneRepositoryViewImpl> {
+
+    }
+
+    private Presenter presenter;
+
+    private static CloneRepositoryFormBinder uiBinder = GWT.create( CloneRepositoryFormBinder.class );
+
+    @UiField
+    Button clone;
+
+    @UiField
+    Button cancel;
+
+    @UiField
+    ControlGroup organizationalUnitGroup;
+
+    @UiField
+    ListBox organizationalUnitDropdown;
+
+    @UiField
+    HelpInline organizationalUnitHelpInline;
+
+    @UiField
+    ControlGroup nameGroup;
+
+    @UiField
+    TextBox nameTextBox;
+
+    @UiField
+    HelpInline nameHelpInline;
+
+    @UiField
+    ControlGroup urlGroup;
+
+    @UiField
+    TextBox gitURLTextBox;
+
+    @UiField
+    HelpInline urlHelpInline;
+
+    @UiField
+    TextBox usernameTextBox;
+
+    @UiField
+    PasswordTextBox passwordTextBox;
+
+    @UiField
+    BaseModal popup;
```

Comment:
OK, this is a classic wrapping a "popup" in a "popup" mistake.

Your custom widget extends ```PopupPanel``` (which is a popup) and you embed another ```BaseModal``` (another popup implementation) in the Widget's composite content. You then show/hide the embedded ```BaseModal```. You should be able to simplify this by having your Widget extend ```BaseModal``` and remove the wrapped popup.